src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.WorkerThread.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using Internal.Runtime.Augments;
+
+namespace System.Threading
+{
+    internal static partial class ClrThreadPool
+    {
+        internal static class WorkerThread
+        {
+            internal static WaitSubsystem.WaitableObject s_semaphore = WaitSubsystem.WaitableObject.NewSemaphore(0, int.MaxValue);
+
+            private const int TimeoutMs = 20 * 1000;
+            
+            private static void WorkerThreadStart()
+            {
+                RuntimeThread.InitializeThreadPoolThread();
+                // TODO: Event: Worker Thread Start event
+
+                while (true)
+                {
+                    // TODO: Event:  Worker thread wait event
+                    while (WaitSubsystem.Wait(s_semaphore, TimeoutMs, false, true))
+                    {
+                        if(!ThreadPoolWorkQueue.Dispatch())
+                        {
+                            return;
+                        }
+
+                        RuntimeThread.CurrentThread.Priority = ThreadPriority.Normal;
+                        CultureInfo.CurrentCulture = CultureInfo.InstalledUICulture;
+                        CultureInfo.CurrentUICulture = CultureInfo.InstalledUICulture;
+                    }
+                    s_threadAdjustmentLock.Acquire();
+                    ThreadCounts counts = ThreadCounts.VolatileReadCounts(ref s_counts);
+                    while (true)
+                    {
+                        if (counts.numActive == counts.numWorking)
+                        {
+                            s_threadAdjustmentLock.Release();
+                            break;
+                        }
+
+                        ThreadCounts newCounts = counts;
+                        newCounts.numActive--;
+                        newCounts.maxWorking = (short)Math.Max(s_minThreads, Math.Min(newCounts.numActive, newCounts.maxWorking));
```

Comment:
s_minThreads and s_maxThreads should also be short, and SetMinThreads and SetMaxThreads should not allow setting a number greater than the max possible count, which fits in a short